Source/platform/locale.cpp

```diff
@@ -176,12 +176,19 @@ std::vector<std::string> GetLocales()
 
 	CFRelease(languages);
 #else
-	try {
-		std::string locale = std::locale("").name();
-		// strip off any encoding specifier, devX uses UTF8.
-		locales.emplace_back(locale.substr(0, locale.find('.')));
-	} catch (std::runtime_error &e) {
-		LogWarn("Locale detection failed: {}", e.what());
+	const auto from = [&locales](string_view locale) -> bool {
+		const string_view lang = locale.substr(0, locale.find_first_of(".:"));
+		if (lang.empty())
+			return false;
+		locales.emplace_back(std::string(lang));
+		return true;
+	};
+	if (!(from(std::getenv("LANGUAGE")) || from(std::getenv("LANG"))
+#ifdef LC_MESSAGES
+	        || from(setlocale(LC_MESSAGES, nullptr))
+#endif
+	        || from(setlocale(LC_CTYPE, nullptr)))) {
+		LogWarn("Locale detection failed");

```

Comment:
`string_view{std::getenv("UNDEFINED_VAR")}` segfaults due to getenv returning a nullptr. Also we can handle languages a bit better given it's expected to be a list of locale codes with no character class suffix

```suggestion
	char* envVar = std::getenv("LANGUAGE");
	if (envVar != nullptr) {
		languages = envVar;
	} else {
		envVar = std::getenv("LANG");
		if (envVar != nullptr) {
			languages = envVar;
		} else {
#ifdef LC_MESSAGES
			languages = setlocale(LC_MESSAGES, nullptr);
#else
			languages = setlocale(LC_CTYPE, nullptr);
#endif
		}
		languages = languages.substr(0, languages.find_first_of("."));
	}

	while (!languages.empty()) {
		size_t separatorPos = languages.find_first_of(":");
		if (separatorPos != 0)
			locales.emplace_back(std::string(languages.substr(0, separatorPos)));

		if (separatorPos != languages.npos)
			languages.remove_prefix(separatorPos + 1);
		else
			break;
	}
```

Reply:
With LANGUAGES undefined and LANG=en_AU.utf8 I get `en_AU`, if I define languages as "en_GB:zh_CN:ru" I get `en_GB, zh_CN, ru` (haven't tried to slot in LANG after languages, figure one or the other is good enough)

Reply:
On my system I get `LANGUAGE=en_GB:en LANG=en_GB.UTF-8`